sklearn/tree/_criterion.pyx

```diff
@@ -226,9 +236,9 @@ cdef class ClassificationCriterion(Criterion):
         self.weighted_n_right = 0.0
```

Comment:
Unfortunately, classification criteria and regression criteria have a different number of parameters in their __cinit__ method, so they can't share one. A possibility is to put a dummy parameter in the regression one, with an unused default value. What do you think?